File: 3794. Reverse String Prefix.py
# Solution of LeetCode Problem 
# 3794. Reverse String Prefix
# Solution in Python 

# Approach 1 (using slicing)

class Solution(object):
    def reversePrefix(self, s, k):
        if k == 1 :
            return  s 
        return s[k-1 :  :-1] + s [k::]

File: 3794. Reverse String Prefix.cpp
// Solution of LeetCode Problem
// 3794. Reverse String Prefix
// Solution in CPP

// Approach - 1

class Solution {
public:
    string reversePrefix(string s, int k) {
        string ans = "" ;
        for(int i = k - 1 ; i >= 0 ; i--)
        {
            ans += s[i];
        }
        for(int i = k ; i < s.size() ; i++)
        {
            ans += s[i];
        }
        return ans;
    }
};
